torch/_inductor/graph.py

```diff
@@ -505,17 +507,25 @@ def allocate():
                     and data.device == value.device
                     and torch.eq(data, value).all()
                 ):
-                    return name
-            name = f"constant{len(self.constants)}"
+                    return constant_name
+
+            if name is None:
+                name = f"constant{len(self.constants)}"
             self.constants[name] = data
             self.constant_reprs[name] = hashlib.sha256(
                 repr(data).encode("utf-8")
             ).hexdigest()
             return name
 
+        name = allocate(name)
+        tensor = data.contiguous()
```

Comment:
same thing here, we shouldnt need to call `.contiguous()`, and now the `tensor` has different  strides than `self.static_sizes_strides(data)`